pkg/scheduler/apis/config/v1/conversion.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"unsafe"
+
+	"k8s.io/apimachinery/pkg/conversion"
+	v1 "k8s.io/kube-scheduler/config/v1"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+)
+
+func Convert_v1_LegacyExtender_To_config_Extender(in *v1.LegacyExtender, out *config.Extender, s conversion.Scope) error {
+	out.URLPrefix = in.URLPrefix
+	out.FilterVerb = in.FilterVerb
+	out.PreemptVerb = in.PreemptVerb
+	out.PrioritizeVerb = in.PrioritizeVerb
+	out.Weight = in.Weight
+	out.BindVerb = in.BindVerb
+	out.EnableHTTPS = in.EnableHTTPS
+	out.TLSConfig = (*config.ExtenderTLSConfig)(unsafe.Pointer(in.TLSConfig))
+	out.HTTPTimeout.Duration = in.HTTPTimeout
+	out.NodeCacheCapable = in.NodeCacheCapable
+	out.ManagedResources = *(*[]config.ExtenderManagedResource)(unsafe.Pointer(&in.ManagedResources))
```

Comment:
having unsafe pointer conversions in this in non-generated code is not great... the generated code runs generate-time checks to make sure the structs are identical... this will break if a field is added to config.ExtenderManagedResource

Reply:
There is still the generated `autoConvert_v1beta1_Extender_To_config_Extender` so the compatibility of both `ManagedResources` and `ExtenderTLSConfig` is still checked. We still get the warning:
```
E0623 11:10:09.813915   49690 conversion.go:755] Warning: could not find nor generate a final Conversion function for k8s.io/kubernetes/pkg/scheduler/apis/config.ExtenderManagedResource -> k8s.io/kube-scheduler/config/v1.ExtenderManagedResource
E0623 11:10:09.814117   49690 conversion.go:756]   the following fields need manual conversion:
E0623 11:10:09.814122   49690 conversion.go:758]       - NewField

```

Do you think this is not enough?

Reply:
Let's replace this with a loop and a call to `autoConvert_v1_ExtenderManagedResource_To_config_ExtenderManagedResource`
And same for reverse conversion.

Reply:
Sorry, not that, but the exported function `Convert_v1_ExtenderManagedResource_To_config_ExtenderManagedResource`

Reply:
Done